### PR TITLE
added vim compatibility, backspace above enter, and general macOS opt…

### PR DESCRIPTION
**Pull request by keithlo**
…mizations on top of default layout

## Description
added vim friendly navigation (little to no travel) and macOS general enhancements on top of the default layout

## Types of Changes
simple keymap addition named vimBack after Vim + backspace 

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
In general, we like the keymap folder name to match the github username, so we can easily identify the ownership of the kemap. 

If the name could be moved, that would be appreciated.

**Comment by ridingqwerty** [comment]
@keithlo any chance you'd be able to get the keymap name updated as drashna requested?

No rush or anything, I'm just going through low-hanging fruit right now and this looks good to merge besides that.

File: keyboards/projectkb/alice/vimBack/keymap.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  [_BASE] = LAYOUT_default(
+    KC_ESC,  KC_TILD, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,   KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_DEL,
+    KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+    KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,             KC_H,   KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+             KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_B,   KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_RGUI,
+             KC_LCTL,          KC_LCMD, KC_SPC,  MO(_FN1),                          KC_RCMD,           KC_RALT,          KC_RCTL
+  ),
+
+  [_FN1] = LAYOUT_default(
+    RGB_TOG,  _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC__VOLDOWN, KC__VOLUP, _______, KC__MUTE,
```
Comment (fauxpark) [inline keyboards/projectkb/alice/vimBack/keymap.c]:
```suggestion
    RGB_TOG,  _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_VOLD, KC_VOLU, _______, KC_MUTE,
```

These keycodes will work on any OS.

File: keyboards/projectkb/alice/vimBack/keymap.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#define _BASE 0
+#define _FN1 1
+
+/*
+ Keycode Legend (macOS)
+ self explanitory:
+ KC_LCMD, KC_LEFT, KC_DOWN, KC_UP, KC_RGHT
+ Volume:
+ KC__MUTE, KC__VOLUP, KC__VOLDOWN
+*/
```
Comment (fauxpark) [inline keyboards/projectkb/alice/vimBack/keymap.c]: This comment can be deleted.

File: keyboards/projectkb/alice/keymaps/keithlo/keymap.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  [_BASE] = LAYOUT_default(
+    KC_ESC,  KC_TILD, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,   KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_DEL,
+    KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+    KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,             KC_H,   KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+             KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,             KC_B,   KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_RGUI,
+             KC_LCTL,          KC_LCMD, KC_SPC,  MO(_FN1),                          KC_RCMD,           KC_RALT,          KC_RCTL
+  ),
+
+  [_FN1] = LAYOUT_default(
+    RGB_TOG,  _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC__VOLD, KC__VOLU, _______, KC_MUTE,
```
Comment (noroadsleft) [inline keyboards/projectkb/alice/keymaps/keithlo/keymap.c:37]: The Volume keycodes here are wrong.

```suggestion
    RGB_TOG,  _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_VOLD, KC_VOLU, _______, KC_MUTE,
```

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks!

**Pull request merged**